internal/checker/checker.go

```diff
@@ -13576,7 +13576,27 @@ func (c *Checker) getTargetOfModuleDefault(moduleSymbol *ast.Symbol, node *ast.N
 }
 
 func (c *Checker) reportNonDefaultExport(moduleSymbol *ast.Symbol, node *ast.Node) {
-	// !!!
+	if moduleSymbol.Exports != nil && moduleSymbol.Exports[ast.InternalSymbolNameExportEquals] != nil {
```

Comment:
This condition might be wrong; I'd expect `decoratorMetadataWithImportDeclarationNameCollision4` to pass.